Tactics/CSE.lean

```diff
@@ -104,6 +104,16 @@ structure CSEConfig where
   minOccsToCSE : Nat := 2
   /-- Whether the tactic should throw an error if no CSEable subterms were found. -/
   failIfUnchanged : Bool := true
+  /-- Number of steps the tactic should spend searching for subterms to gather information. -/
+  fuelSearch : Nat := 1000
+  /-- Number of steps the tactic should spend performing subexpression elimination.
+   It can be useful to have large amounts of fuel for searching, and very little for eliminating,
+   to search for maximal subterm sharing, and to then eliminiate the most common occurrences.
+  -/
+  fuelEliminate : Nat := 1000
+  /-- Whether we are performing a dry run, where we will identify repeated
+      subterms, and simulate the sharing algorithm, but perform no actual sharing. -/
+  dryRun? : Bool := false
```

Comment:
Add user-guidance for when to use `dryRun?`, and how it reports the info. it gleans.

Reply:
@shigoel I wrote that this meant for debugging. I'd rather not write down how it reports the info, since I could see that evolving with the tactic --- the implementation essentially writes down what it skipped due to it being a dry run.

Reply:
Alright -- that works.